src/cli/rustup_mode.rs

```diff
@@ -404,6 +405,13 @@ pub fn cli() -> App<'static, 'static> {
                     SubCommand::with_name("uninstall")
                         .about("Uninstall a toolchain")
                         .alias("remove")
+                        .arg(
+                            Arg::with_name("pattern")
+                                .help("Treat arguments as glob patterns")
+                                .short("p")
+                                .long("pattern")
+                                .takes_value(false),
+                        )
```

Comment:
We have a couple options here:

1. **Treat all toolchains as patterns.** Upside: simpler to use. Downside: could be confusing if that's not what people expected.

2. **Only treat toolchains with metacharacters (`*` or `?` I think).** Upside: also simpler to use. Downside: we'd have to figure out a way to make sure we catch edge-cases, etc.

3. **Keep it as an optional behavior enabled with a flag.** We may want to bikeshed whether this should be `--pattern` or `--glob`. Perhaps look at prior art?

Reply:
If people currently put shell glob characters into their toolchain names then they're probably in a world of pain without us treating all incoming names as patterns, so my vote is for 1.  @rbtcollins what do you think?

Reply:
depends on the patterns we're talking about . '.' for instance from regex patterns is present in version numbers. If we're just talking '*' and '?' as pattern characters, then I think we're fine.

Reply:
The docs for the glob crate say:

- `?` matches any single character.

- `*` matches any (possibly empty) sequence of characters.

- `**` matches the current directory and arbitrary subdirectories. This
  sequence **must** form a single path component, so both `**a` and `b**`
  are invalid and will result in an error.  A sequence of more than two
  consecutive `*` characters is also invalid.

- `[...]` matches any character inside the brackets.  Character sequences
  can also specify ranges of characters, as ordered by Unicode, so e.g.
  `[0-9]` specifies any character between 0 and 9 inclusive. An unclosed
  bracket is invalid.

- `[!...]` is the negation of `[...]`, i.e. it matches any characters
  **not** in the brackets.

- The metacharacters `?`, `*`, `[`, `]` can be matched by using brackets
  (e.g. `[?]`).  When a `]` occurs immediately following `[` or `[!` then it
  is interpreted as being part of, rather then ending, the character set, so
  `]` and NOT `]` can be matched by `[]]` and `[!]]` respectively.  The `-`
  character can be specified inside a character sequence pattern by placing
  it at the start or the end, e.g. `[abc-]`.

So I think it's fine to go with option 1.